### PR TITLE
Add sum-fibs program

### LINKED ISSUE #5: Add sum-fibs program

### PR DESCRIPTION
**Pull request by vaibhavarora14**
**Program strategy -**
- Get `fibonacci numbers` starting from 1 in for loop, and apply `addition` operation on them for `total sum`.
- When `fiboancci number` is higher than program input, for loop ends and `total sum` is returned from program

### CODE REVIEW

File: js-exercises/sum-fibs/sumFibs.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+const isNumber = (arg) => typeof arg === 'number';
+
+const isOddNumber = (arg) => isNumber(arg) && (arg % 2 !== 0);
+
+const isPositiveNumber = (arg) => isNumber(arg) && (arg >= 0);
+
+function* returnNumberGenerator(number) {
```
Comment (ArfatSalman) [inline js-exercises/sum-fibs/sumFibs.js]: Why this? Since you've used this in only one place.

File: js-exercises/sum-fibs/sumFibs.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+const isNumber = (arg) => typeof arg === 'number';
+
+const isOddNumber = (arg) => isNumber(arg) && (arg % 2 !== 0);
+
+const isPositiveNumber = (arg) => isNumber(arg) && (arg >= 0);
```
Comment (ArfatSalman) [inline js-exercises/sum-fibs/sumFibs.js]: 0 is neither positive not negative. https://math.stackexchange.com/questions/26705/is-zero-positive-or-negative

File: js-exercises/sum-fibs/sumFibs.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+const isNumber = (arg) => typeof arg === 'number';
+
+const isOddNumber = (arg) => isNumber(arg) && (arg % 2 !== 0);
```
Comment (ArfatSalman) [inline js-exercises/sum-fibs/sumFibs.js:3]: This would be true for `NaN`

File: js-exercises/sum-fibs/sumFibs.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  if (endNumber < 1) {
+    throw Error(`Fibonnaci series starts with 1, and param sent - ${endNumber} is less than 1`);
+  }
+
+  const startNumber = 1;
+  let result = startNumber;
+
+  const fibonacciIterator = fibonacciGenerator(startNumber);
+
+  for (const number of fibonacciIterator) {
```
Comment (ArfatSalman) [inline js-exercises/sum-fibs/sumFibs.js]: You can also create a `filter` generator like and put the pass the isOddNumber there to only generate odd Fibonacci numbers.

File: js-exercises/sum-fibs/sumFibs.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    if (number > endNumber) {
+      break;
```
Comment (ArfatSalman) [inline js-exercises/sum-fibs/sumFibs.js]: Can you do this without `break`?

**Review by ArfatSalman — CHANGES_REQUESTED** [review]
This is not a very good solution. You can do better by creating a Fibonacci number generator, a "filter" generator and pass the filtering condition to the `filter` function. Then you can only `take` n number of Fib numbers.

**Comment by ArfatSalman** [comment]
Also, you can try using libraries like `lodash` to not write common utils again and again.

File: js-exercises/sum-fibs/sumFibs.js (hunk 1 above)
Comment (vaibhavarora14) [reply js-exercises/sum-fibs/sumFibs.js]: It was cool for me to use `destructuring assignment` with help of generator.
Now it's not making that much sense, and I will remove this.

Comment (vaibhavarora14) [reply js-exercises/sum-fibs/sumFibs.js]: Removed with 8ab346c55474d57acbd63b0f97940c49cb53006b

File: js-exercises/sum-fibs/sumFibs.js (hunk 4 above)
Comment (vaibhavarora14) [reply js-exercises/sum-fibs/sumFibs.js]: Maybe 8ab346c55474d57acbd63b0f97940c49cb53006b resolves the problem in context you pointed.
I am not sure, so please reply after viewing the changes 🙂

File: js-exercises/sum-fibs/sumFibs.js (hunk 5 above)
Comment (vaibhavarora14) [reply js-exercises/sum-fibs/sumFibs.js]: `break` is removed with 8ab346c55474d57acbd63b0f97940c49cb53006b

File: js-exercises/sum-fibs/sumFibs.js (hunk 3 above)
Comment (vaibhavarora14) [reply js-exercises/sum-fibs/sumFibs.js:3]: right, fixed issue with bcb3526112c874b7365fbbf8a957a7479e2f727c

File: js-exercises/sum-fibs/sumFibs.js (hunk 2 above)
Comment (vaibhavarora14) [reply js-exercises/sum-fibs/sumFibs.js]: right, code is updated with 3dd02cd49ac897ea7001e3ebeb1253592843577a

**Comment by vaibhavarora14** [comment]
@ArfatSalman, I am not going to use `loadash` very easily. 😀
I will write utilities, fix issue there, learn from them.
I am re-using those utilities by copy-paste.

Once code is in master, you will find utilities spread and re-used from code.

**Pull request closed without merging**